### ISSUE TITLE
PageCoordinator?

### ISSUE THREAD

**Issue by AAAstorga** [quickbirdstudios/XCoordinator-Example#3]
I am confused on how PageCoordinator works in this example. When the HomePageCoordinator is shown it just shows the 3 buttons (logout, users about). Is there supposed to be a swipeable page controller?

I am looking to adopt XCoordinator in my app, but I cannot understand how to use a simple PageCoordinator for an onboarding flow. [Screen1, Screen2, Login Screen] as the pages. Any guidance would be great!

Thank you! :)

**Comment by pauljohanneskraft** [comment]
Keep in mind, that whenever you tap the loginButton, one of the following coordinators is shown: `HomeTabBarCoordinator`, `HomeSplitCoordinator` and `HomePageCoordinator`. To change this behavior, please adapt the `prepareTransition` method in the `AppCoordinator` to your liking (e.g. always showing a `HomePageCoordinator`).

Here are some basics about PageCoordinators starting with the initializers of a PageCoordinator:

Option 1:
```swift
public init(rootViewController: RootViewController = .init(),
    pages: [Presentable],
    loop: Bool = false,
    set: Presentable? = nil,
    direction: UIPageViewController.NavigationDirection = .forward)
```

Option 2:
```swift
public init(rootViewController: RootViewController = .init(),
    dataSource: UIPageViewControllerDataSource,
    set: Presentable,
    direction: UIPageViewController.NavigationDirection)
```

Option 1 offers a default UIPageViewControllerDataSource, which gets initialized with an array of presentables (being either coordinator or viewController). You can initialize the PageCoordinator using that initializer for a fixed array of presentables, e.g. in the onboarding case you discussed - however, if you need more control about the presentables you want to use in the UIPageViewController, you may use Option 2.

The `Transition.set` can be used to set the currently visible pages, which can either be one Presentable or two, if you set isDoubleSided to true. For further customization, you can access the UIPageViewController using the `rootViewController` property in your coordinator.

I hope, I could help you with setting up a PageCoordinator.

**Comment by pauljohanneskraft** [comment]
I actually noticed a problem with the current implementation of PageCoordinator, which should be fixed with https://github.com/quickbirdstudios/XCoordinator/pull/134. Until then, you can override the initializer of your PageCoordinator subclass and call `rootViewController.dataSource = self.dataSource` directly after `super.init`, which should solve the problem - thanks for catching that one!

**Comment by AAAstorga** [comment]
Thank you for putting out a fix for this! Glad I could help. Thanks for the great library!

**Comment by pauljohanneskraft** [comment]
The fix was part of the 2.0.5 release - so I'm closing this now. 
Glad you like it and thanks again for your help.
